Payout activity panel

donalnofrixion opened the pull request.

donalnofrixion commented:
https://github.com/nofrixion/nofrixion.business/assets/89344328/f8bc79c8-fe9a-4d94-9781-2a70e5eecfc0

sipsorcery commented:
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/144.

sipsorcery commented:
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/144.

sipsorcery commented:
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/144.

sipsorcery commented:
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/144.

donalnofrixion commented:
Hi @pabloi-nf I've made those changes we discussed.

1. Added `Edited by`
2. Removed `Unknown` activity
3. Updated activity text. `Waiting for bank's authorisation` & `Submitted for bank's authorisation`

sipsorcery commented:
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/144.

Publicker commented on packages/components/src/components/ui/atoms/Icon/Icon.tsx:
```diff
@@ -766,6 +766,35 @@ export const Icons = {
       </defs>
     </SVG>
   ),
+  'failed-red/16': (className: string) => (
+    <SVG className={cn(className, 'fill-current stroke-none')} size="16">
+      <circle cx="8" cy="8" r="8" fill="#F32448" />
+      <path
+        fillRule="evenodd"
+        clipRule="evenodd"
+        d="M6.20712 4.7929C5.8166 4.40237 5.18343 4.40236 4.7929 4.79288C4.40237 5.1834 4.40236 5.81657 4.79288 6.2071L6.58667 8.00093L4.7931 9.79455C4.40258 10.1851 4.40259 10.8182 4.79312 11.2088C5.18365 11.5993 5.81681 11.5993 6.20733 11.2087L8.00087 9.41517L9.7944 11.2087C10.1849 11.5993 10.8181 11.5993 11.2086 11.2088C11.5991 10.8182 11.5992 10.1851 11.2086 9.79455L9.41506 8.00093L11.2089 6. ...
+        fill="white"
+      />
+    </SVG>
+  ),
+  'created/16': (className: string) => (
+    <SVG className={cn(className, 'fill-current stroke-none')} size="16">
+      <circle cx="8" cy="8" r="8" fill="#454D54" />
+      <circle cx="8.00008" cy="8.00008" r="3.33333" fill="white" />
+    </SVG>
+  ),
+  'success/16': (className: string) => (
+    <SVG className={cn(className, 'fill-none stroke-none')} size="16">
+      <circle cx="8" cy="8" r="8" fill="#29A37A" />
+      <path
+        d="M4.25 8.26316L6.61842 10.6316L11.75 5.5"
+        stroke="white"
+        strokeWidth="2"
+        strokeLinecap="round"
+        strokeLinejoin="round"
+      />
+    </SVG>
+  ),
```
Can we isolate and remove the hardcoded fill and stroke colours from this? I know we can manage strokes by setting the font color externally, but I'm not sure about the fills.

donalnofrixion replied:
Not sure TBH. I left them in here because they are colours that probably won't change. Red for failed, green for success, etc

Publicker replied:
Okay, makes sense.

pabloi-nf replied:
I'm not sure if it applies, but even when they are probably still going to be "green" and "red", the specific shade can vary in a future redesign, or maybe if we allow for user customization the "success" and "failure" colors could be adapted by the UI component user. 
These are 2 not very likely scenarios, but just to keep it in mind

sipsorcery commented:
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/144.

donalnofrixion commented:
Hi @pabloi-nf The rule name will now be displayed if the payout was automatically created.
I'll create a bug for the duplicate paid event in the api. This will be a separate PR

![image](https://github.com/nofrixion/nofrixion.business/assets/89344328/b4460fe4-7485-4cc5-96d5-f84425b9052c)
